modules/uhsg_active_degree_programme/src/PageCache/DenyActiveDegreeProgramme.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Drupal\uhsg_active_degree_programme\PageCache;
+
+use Drupal\Core\PageCache\ResponsePolicyInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class DenyActiveDegreeProgramme implements ResponsePolicyInterface {
+
+  /**
+   * @inheritdoc
+   */
+  public function check(Response $response, Request $request) {
```

Comment:
After investigating to find less aggressive caching, we found out that proposed solution works with dynamic page caching sufficient enough to pass this through to upstream. There are also other features/modules that potentially can kill performance.

We'll measure and find ways to optimize caching when we get upstream to full-feature state. We also need to specify the service requirements in terms of traffic so that, we can provide realistic solutions with the required feature-set and traffic requirements.